### ISSUE TITLE
DateInput is not updating the value of the DatePickerDialog

### ISSUE THREAD

**Issue by tuantvu** [saas-js/saas-ui#206]
### Description

When using the DateInput or DatePickerInput with a DatePickerDialog, on change of the DateInput or DatePickerInput does not update the value in the DatePickerDialog. This can be recreate-able in the Saas-UI live demo. Consistently keeping the values in sync between the two components feels more user friendly.

### Link to Reproduction

[https://saas-ui.dev/docs/components/date-time/date-input](https://saas-ui.dev/docs/components/date-time/date-input)

### Steps to reproduce

1. Go to any live demo example
2. Edit the input date field to another month or year
3. Open the date picker dialog using the calendar icon
4. The date picker dialog will still show the current month and year

### Saas UI Version

0\.31.1

### Chakra UI Version

2\.2.0

### Browser

Chrome 121.0.6167.184

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

![Screenshot 2024-02-19 at 1 57 29 PM](https://uploads.linear.app/14807ff2-55c5-44c9-9588-135c961250cc/b77f010b-0f99-4e73-8b0c-c3191cc55c52/3369903c-f49e-4b80-96e0-c9b75facdf00?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzE0ODA3ZmYyLTU1YzUtNDRjOS05NTg4LTEzNWM5NjEyNTBjYy9iNzdmMDEwYi0wZjk5LTRlNzMtOGIwYy1jMzE5MWNjNTVjNTIvMzM2OTkwM2MtZjQ5ZS00YjgwLTk2ZTAtYzliNzVmYWNkZjAwIiwiaWF0IjoxNzA5MTI1ODkxLCJleHAiOjMzMjc5Njg1ODkxfQ.95FG43VrAKp5R5ZVwSm1MTCWOHIXqkA4y5F8xzQRN88)

**Comment by tuantvu** [comment]
I stumbled upon a workaround. I don't know why, but when I extracted the DatePickerDialog into a private component, the DatePickerInput and DatePickerDialog stays in sync. Using the DatePickerDialog or wrapping it with one layer of Box doesn't work. See example below.

```
const MyComponent = () => {
    // private component
    const Dialog = () => (
      <DatePickerDialog>
        <DatePickerCalendar />
      </DatePickerDialog>
    )

    return (
      <DatePicker
        placement="bottom"
        granularity="day"
      >
       
          <DatePickerInput />

        {/* WORKS*/}
        <Dialog />

        {/* DOES NOT WORK
          <DatePickerDialog>
            <DatePickerCalendar />
          </DatePickerDialog>
        */}

         {/* ALSO DOES NOT WORK
         <Box>
          <DatePickerDialog>
            <DatePickerCalendar />
          </DatePickerDialog>
        </Box> */}
      </DatePicker>
    )
}
```

**Comment by Pagebakers** [comment]
Interesting that this fixes it, thanks for the info. I'm going to investigate this now.

**Comment by Pagebakers** [comment]
The value is actually in sync, but the calendar doesn't open on the correct month.

**Comment by Pagebakers** [comment]
@tuantvu it seems like a limitation of react aria calendar. Found a fix, will release it later this week.

**Comment by Pagebakers** [comment]
This was released in 0.12.23
